src/GraphQL.ApiTests/GraphQL.approved.txt

```diff
@@ -1985,6 +1989,11 @@ namespace GraphQL.Types
         GraphQL.Language.AST.IValue Convert(object value, GraphQL.Types.IGraphType type);
         bool Matches(object value, GraphQL.Types.IGraphType type);
     }
+    public interface ICanParseScalar
+    {
+        bool CanParseLiteral(GraphQL.Language.AST.IValue value);
+        bool CanParseValue(object value);
+    }
```

Comment:
Why don't we just require these methods on `ScalarGraphType` ?  We can include a default implementation as follows:

```cs
public virtual CanParseLiteral(IValue value)
{
    try {
        return ParseLiteral(value) != null;
    } catch {
        return false;
    }
}
```

That's basically what happens now anyway within `IsValidLiteralValue`.

(And similar of course for `CanParseValue`).

Reply:
Obviously I didn't think about it. Done.